src/thunderbots/software/ai/ai_wrapper.cpp

```diff
@@ -0,0 +1,63 @@
+#include "ai/ai_wrapper.h"
+
+#include <boost/bind.hpp>
+
+#include "thunderbots_msgs/PlayInfo.h"
+#include "util/canvas_messenger/canvas_messenger.h"
+#include "util/parameter/dynamic_parameters.h"
+
+AIWrapper::AIWrapper(ros::NodeHandle node_handle)
+{
+    play_info_publisher = node_handle.advertise<thunderbots_msgs::PlayInfo>(
+        Util::Constants::PLAY_INFO_TOPIC, 1);
+}
+
+void AIWrapper::onValueReceived(World world)
+{
+    most_recent_world = world;
+    runAIAndSendPrimitives();
+    publishPlayInfo();
+    drawWorld();
+}
+
+void AIWrapper::publishPlayInfo()
+{
+    auto play_info_msg = convertPlayInfoToROSMessage(ai.getPlayInfo());
+    play_info_publisher.publish(play_info_msg);
+}
+
+void AIWrapper::runAIAndSendPrimitives()
+{
+    if (Util::DynamicParameters::AI::run_ai.value())
```

Comment:
Not quite sure what you mean, there is no `has_value()` function here?

Reply:
To clarify here, @EvanMorcom the `run_ai` variable here is *not* an `std::optional`, it is our own `Parameter` type. There is no `has_value` function, `value` just gets the value of the parameter.

Reply:
Changing the function to something like `getValue` (which I would be on board with since it would match the `setValue` function) should be handled in a separate PR. @akhilveeraghanta 	FYI

Reply:
Right, I wasn't very clear here. It's not immediately clear whether run_ai.value() is. It could be 1/0 true/false, and assuming cpp 'if' statements are similar to c it could be any integer value.

Very minor detail, I'm totally fine if we ignore this.

Reply:
I think it's out of scope for this PR. We can change how we get values from the `Parameter` class, but it's going to be a generic call anyway so there won't be different ones depending on the Parameter type (like a different one for `Parameter<bool>` vs `Parameter<double>`